src/cephadm/cephadm

```diff
@@ -765,9 +765,10 @@ def call(command,  # type: List[str]
         end_time = start_time + timeout
     while not stop:
         if end_time and (time.time() >= end_time):
-            logger.info(desc + ':timeout after %s seconds' % timeout)
             stop = True
-            process.kill()
+            if process.poll() is None:
```

Comment:
sure about this? I suspect we also have to kill the process, if `poll()` is not none. I mean, calling `poll()` is useless here, as this would be racy with the process.

Reply:
It's racy in either form ... but I don't see how issuing another SIGKILL will help when the process has already exited and the loop is simply busy reading the remaining buffers from stdout/stderr .. ?

Reply:
Sigh, this loop is so irritating and complex. 

~~ok, what about.~~

1. ~~in line 768, we set `stop = True`~~
1. ~~in line 769 and `process.poll() is not None`, i.e. the process is still alive.~~
1. ~~ in line 772, `process.poll() is None`~~

never mind. 

> Check if child process has terminated. Set and return returncode attribute. Otherwise, returns None.